Sources/Get/APIClientDelegate.swift

```diff
@@ -57,6 +57,27 @@ public protocol APIClientDelegate {
     /// - returns: The URL for the request. Return `nil` to use the default
     /// logic used by client.
     func client<T>(_ client: APIClient, makeURLForRequest request: Request<T>) throws -> URL?
+
+    /// Allows you to override the client's encoder for a specific request.
+    ///
+    /// If not implemented, the request uses the client's encoder.
+    ///
+    /// - Parameters:
+    ///   - client: The client that sends the request.
+    ///   - request: The request about to be sent.
+    /// - Returns: The JSONEncoder for the request.
+    func client<T>(_ client: APIClient, encoderForRequest request: Request<T>) -> JSONEncoder
```

Comment:
Alternatively, these delegates could return an optional to avoid the user needing to pluck out the underlying decoder/encoder in cases they don't want to override.
If the delegate returns nil, use the client's default decoder/encoder.

Reply:
I updated the delegates to return an optional